File: src/main/java/minic/ui/MiniCAssemblyTextModelFactory.java
package minic.ui;

import minic.uiapi.UiStageVisualDto;

import java.util.List;

/**
 * 根据 Assembly visual DTO 生成等宽文本行模型。
 */
public final class MiniCAssemblyTextModelFactory {
    /**
     * 创建 Assembly 文本行。
     *
     * @param visual 当前阶段 visual DTO
     * @return 文本行
     */
    public List<MiniCAssemblyTextLine> create(UiStageVisualDto visual) {
        return visual.assemblyLines().stream()
                .map(line -> new MiniCAssemblyTextLine(
                        line.lineNumber(),
                        line.text(),
                        line.section(),
                        line.label(),
                        line.kind(),
                        line.range(),
                        line.active()
                ))
                .toList();
    }
}

File: uiweb/src/visual/MiniCAssemblyTextModelFactory.ts
import type { JavaMirrorFile } from "../translation/javaMirror";
import type { UiStageVisualDto } from "../translation/uiapi";
import { MiniCAssemblyTextLine } from "./MiniCAssemblyTextLine";

export const miniCAssemblyTextModelFactoryMirror = {
  javaPath: "src/main/java/minic/uilocal/visual/MiniCAssemblyTextModelFactory.java",
  webPath: "uiweb/src/visual/MiniCAssemblyTextModelFactory.ts",
  packageName: "minic.uilocal.visual",
  exportName: "MiniCAssemblyTextModelFactory",
  kind: "class",
  imports: [
    "java.util.List",
    "minic.uiapi.UiStageVisualDto"
  ],
  fields: [],
  methods: [
    {
      "name": "create",
      "signature": "create(UiStageVisualDto visual)"
    }
  ],
} as const satisfies JavaMirrorFile;

export class MiniCAssemblyTextModelFactory {
  static readonly mirror = miniCAssemblyTextModelFactoryMirror;

  create(visual: UiStageVisualDto | null | undefined): readonly MiniCAssemblyTextLine[] {
    return (visual?.assemblyLines ?? []).map((line) => new MiniCAssemblyTextLine(
      line.lineNumber,
      line.text,
      line.section,
      line.label,
      line.kind,
      line.range,
      line.active,
    ));
  }
}

export default MiniCAssemblyTextModelFactory;
